test/e2e/cmd/run/run_test.go

```diff
@@ -0,0 +1,155 @@
+	return &FakeLogStream{
+		FakeLogStreamProvider: f,
+		pos:                   0,
+		reader:                bytes.NewReader(f.data),
+	}, nil
+}
+
+func (f *FakeLogStreamProvider) String() string {
+	return "test_pod"
+}
+
+// Read reads the underlying bytes or returns an error when half of the stream has been sent
+func (us *FakeLogStream) Read(p []byte) (int, error) {
+	n, err := us.reader.Read(p)
+	us.pos += n
+	if us.withError && !us.failed && us.pos > len(us.data)/2 {
+		us.failed = true
+		return 0, fmt.Errorf("test error")
+	}
+	if err == io.EOF {
+		// informs streamTestJobOutput that the end of the stream has been reached
+		close(us.stop)
+	}
+	return n, err
+}
+
+func (us *FakeLogStream) Close() error {
+	// noop
+	return nil
+}
+
+// Test_helper_streamTestJobOutput_withError simulates an error while the stream is read using the FakeLogStreamProvider
+func Test_helper_streamTestJobOutput_withError(t *testing.T) {
+
+	log = logf.Log.WithName("streamTestJobOutput_withError")
+
+	stopLogStream := make(chan struct{})
+	sampleLogs, err := ioutil.ReadFile("testdata/stream.json")
+	require.NoError(t, err)
+	streamProvider := NewFakeLogStreamProvider(sampleLogs, stopLogStream, true)
+
+	streamErrors := make(chan error, 4096)
+	writer := bytes.NewBuffer([]byte{})
+	streamTestJobOutput(streamProvider, writer, streamErrors, stopLogStream)
+
+	got, err := ioutil.ReadAll(writer)
+	require.NoError(t, err)
+
+	// Check that we had one error
+	errorCount := len(streamErrors)
+	assert.Equal(t, 1, errorCount)
+
+	// Check that the data are the expected ones
+	assert.Equal(t, sampleLogs, got)
+}
+
+func Test_helper_streamTestJobOutput(t *testing.T) {
+
+	log = logf.Log.WithName("streamTestJobOutput")
+
+	stopLogStream := make(chan struct{})
+	sampleLogs, err := ioutil.ReadFile("testdata/stream.json")
+	require.NoError(t, err)
+	streamProvider := NewFakeLogStreamProvider(sampleLogs, stopLogStream, false)
+
+	streamErrors := make(chan error, 4096)
+	writer := bytes.NewBuffer([]byte{})
+	streamTestJobOutput(streamProvider, writer, streamErrors, stopLogStream)
+
+	// Check that the data written are the
+	got, err := ioutil.ReadAll(writer)
+	require.NoError(t, err)
+
+	errorCount := len(streamErrors)
+	assert.Equal(t, 0, errorCount)
+
+	assert.Equal(t, sampleLogs, got)
+}
+
+func Test_parseLog(t *testing.T) {
+	type args struct {
+		line string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    time.Time
+		wantErr bool
+	}{
+		{
+			name: "happy path",
+			args: args{
+				line: `{"Time":"2020-08-19T07:55:30.02987855Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/test/e2e/beat","Test":"TestBeatKibanaRefWithTLSDisabled/All_expected_Pods_should_eventually_be_ready","Output":"=== RUN   TestBeatKibanaRefWithTLSDisabled/All_expected_Pods_should_eventually_be_ready\n"}`,
+			},
+			want: time.Date(2020, time.August, 19, 07, 55, 30, 29878550, time.UTC),
+		},
+	}
```

Comment:
for the sake of completeness it would be nice to add another small example with an error (garbage non-JSON string).